internal/gengapic/gengapic.go

```diff
@@ -348,6 +348,36 @@ func (g *generator) insertRequestHeaders(m *descriptor.MethodDescriptorProto, t
 	}
 }
 
+// insertAutoPopulatedFields generates the conditional initialization of any
+// default-value request fields (for the given method) that are specified for
+// auto-population by autoPopulatedFields.
+//
+// If the field value is not equal to default value at the time of sending the
+// request, implying it was set by the user, or if the field has explicit
+// presence and is set by the user, the field must not be auto-populated by
+// the client. Values automatically populated this way must be reused for
+// retries of the same request.
+func (g *generator) initializeAutoPopulatedFields(servName string, m *descriptor.MethodDescriptorProto) {
+	apfs := g.autoPopulatedFields(servName, m)
+	if len(apfs) == 0 {
+		return
+	}
+	g.imports[pbinfo.ImportSpec{Path: "github.com/google/uuid"}] = true
+	p := g.printf
+	for _, apf := range apfs {
+		f := snakeToCamel(apf.GetName())
+		if apf.GetProto3Optional() {
+			// Type will be *string if field has explicit presence.
+			p("if req != nil && req.%s == nil {", f)
+		} else {
+			// Type will be string if field does not have explicit presence.
+			p("if req != nil && req.%s == \"\" {", f)
+		}
+		p("  req.%s = uuid.NewString()", f)
```

Comment:
If it is `Proto3Optional` you will need to make this a pointer value e.g. 
`req.%s = proto.String(uuid.NewString())` via the [`proto.String` helper](https://pkg.go.dev/google.golang.org/protobuf@v1.32.0/proto#String)

Reply:
I get:

```
gen/github.com/googleapis/gapic-showcase/client/echo_client.go:682:19: cannot use proto.String(uuid.NewString()) (value of type *string) as string value in assignment
```
Any suggestions?

Reply:
> If it is `Proto3Optional` you will need to make this a pointer value

Only if it's a `Proto3Optional` field :)

Reply:
Fixed, thanks!